src/prefect/agent.py

```diff
@@ -66,19 +57,42 @@ def __init__(
             self.default_infrastructure = Process()
             self.default_infrastructure_document_id = None
 
-    async def work_queue_id_from_name(self) -> Optional[UUID]:
+    async def get_work_queues(self) -> Iterator[WorkQueue]:
         """
-        For agents that were provided a work_queue_name, rather than a work_queue_id,
-        this function will retrieve the work queue ID corresponding to that name.
-        If no matching queue is found, a warning is logged and `None` is returned.
+        Loads the work queue objects corresponding to the agent's target work
+        queues. If any of them don't exist, they are created.
         """
-        if not self.work_queue_name:
-            raise ValueError("No work queue name provided.")
-        try:
-            work_queue = await self.client.read_work_queue_by_name(self.work_queue_name)
-            return work_queue.id
-        except ObjectNotFound:
-            return await self.client.create_work_queue(name=self.work_queue_name)
+
+        # if queues were cached less than 30 seconds ago, yield the cached values
+        now = pendulum.now("UTC")
+        if self._cache.get("ts", now.subtract(seconds=31)) > now.subtract(seconds=30):
```

Comment:
If only we could walrus. This kind of awkward, can we just handle the case where the cache is empty explicitly? e.g.
```suggestion
        if self._cache and self._cache["ts"] > now.subtract(seconds=30):
```

Somewhat related, I have a minor preference for splitting the cache into two separate variables for clarity

```python
self._work_queue_cache_expiration: datetime = None
self._work_queue_cache: List[WorkQueue] = []
```

Reply:
Thanks, great suggestion